src/cryptsetup/cryptsetup-generator.c

```diff
@@ -335,6 +342,31 @@ static int create_disk(
                         fprintf(f, "Requires=%s\n", unit);
         }
 
+        if (hdrdev) {
+                _cleanup_free_ char *unit = NULL, *p = NULL;
+
+                /* header device timeout does not make sense, rw access is required for LUKS2 recovery  */
+                r = generate_dev_mount(name, hdrdev, "hdrdev", NULL, false, false, &unit, &hdrdev_mount);
```

Comment:
es above, please add comments to the booleans saying what they control (also line break please, as above)

Reply:
Fixed